plonk/src/bencher.rs

```diff
@@ -0,0 +1,184 @@
+//! Helper functions for micro-benchmarks
+
+use ark_std::{thread_local, time::Instant};
+use core::{cell::RefCell, time::Duration};
+
+thread_local!(static FFT_START_TIME: RefCell<Instant> = RefCell::new(Instant::now()));
+thread_local!(static FFT_TIMER_LOCK: RefCell<bool> = RefCell::new(false));
+thread_local!(static FFT_TOTAL_TIME: RefCell<Duration> = RefCell::new(Duration::ZERO));
```

Comment:
I'm not sure if there's better method here, basically we are dealing with a _global time counter_ that accumulates time spent across code snippets in many functions. 

1. I wonder if we can get rid of the time lock, and start time -- they can be local variable, right? only the total time needs to be globally accessible.